### PR TITLE
[WIP] A new T4 xenomorph, the xeno CAS equivalent, the dragon

### PR DESCRIPTION
**Pull request by Arturlang**
## About The Pull Request
Adds a new T4 xenomorph, the dragon, a way for the xenos to have their own version of CAS, via it flying and shooting fireballs from the sky.
Design doc: https://docs.google.com/document/d/1s5xo7GvZHubklFhDcCmdpw3DK4Ouz1nX2TGcSQV5aVI
Design by: https://github.com/Starfirejordan
Art by: https://github.com/Inanichts

## Why It's Good For The Game
This adds a way for xenos to have their own forces in the sky, so marines aren't the only ones able to bomb the hell out of their enemies, where xenos did not have much of reprisal against that, this should hopefully change that.
## Changelog
:cl:
add: Added a new t4 caste, the dragon.
add: Adds a flamethrower variant that smooths with other tiles of itself.
code: You can now whitelist what flamethrower fires damage
/:cl:

### CODE REVIEW

**Comment by DeltaFire15** [comment]
However this ends up balance-wise, that sure is a cool sprite.

**Comment by Starfirejordan** [comment]
I am a man of my word, dragon is here

**Comment by Starfirejordan** [comment]
here be Dragon's

**Comment by Ragecrabbo** [comment]
> Requires 11 others xenos to be alive to evolve into
> 
Personally i would increase this to 14 or possibly even as high as 16 so xenos dont get a massive bump in power as soon as they hit ~12ish population and gain access to both king and dragon at the same time. 
Stats seem also pretty high with survivability of a gorger and slash damage that rivals even ravager. 
Fire is also suprisingly good at taking out marines because of its damage over time effect requiring marines to put themselves out by either resisting (terrible idea when surrounded by xenos) or unwielding their weapon and using a fire extinguisher (less terrible idea but also not great when you have xenos already nibbling at your flesh). Not sure how you would make the abilities involving fire less strong (or if there is even need for them to be nerfed) but i would advise lowering atleast the general stats of the caste itself a bit so you cant just run and slash at marines and cause a ton of damage without taking account the additional use of your abilities.

**Comment by Losenis** [comment]
After witnessing the amount of damage to marines CAS can make accidentally, I dread to imagine what a CAS-equivalent can do on purpose

**Comment by Arturlang** [comment]
> > Requires 11 others xenos to be alive to evolve into
> 
> Personally i would increase this to 14 or possibly even as high as 16 so xenos dont get a massive bump in power as soon as they hit ~12ish population and gain access to both king and dragon at the same time. Stats seem also pretty high with survivability of a gorger and slash damage that rivals even ravager Fire is also suprisingly good at taking out marines because of its damage over time effect requiring marines to put themselves out by either resisting (terrible idea when surrounded by xenos) or unwielding their weapon and using a fire extinguisher (less terrible idea but also not great when you have xenos already nibbling at your flesh). Not sure how you would make the abilities involving fire less strong (or if there is even need for them to be nerfed) but i would advise lowering atleast the general stats of the caste itself a bit so you cant just run and slash at marines and cause a ton of damage without taking account the additional use of your abilities.

All the numbers can be easily changed, and aren't really final, so suggestions are super welcome, for now, I'll increase it to 14, and we'll see how it goes.

**Comment by Moggram** [comment]
_The AIM-224 is the latest in air to air missile technology. Earning the nickname of 'Widowmaker'_

SOOOoooo.... The Condor can shoot them down right?

**Comment by Arturlang** [comment]
> _The AIM-224 is the latest in air to air missile technology. Earning the nickname of 'Widowmaker'_
> 
> SOOOoooo.... The Condor can shoot them down right?

TODO, not sure how I'd implement it, ideas?

**Comment by DeltaFire15** [comment]
> TODO, not sure how I'd implement it, ideas?

I'm not very polarized towards the yes or no of the idea, but I imagine one way to implement cas-dragon interactions would be having dragon get hit and grounded by CAS if they're on the tile with the dot while airborne..

Though thats somewhat unlikely to happen if the dragon isn't blind so a bit weak if one wants cas and dragon to ""fight"" eachother.. Although cas having more effective ways to counterplay dragon would also open up the wasp's (dragon's?) nest of it in turn needing to be able to counterplay CAS itself.

**Comment by Coastfront** [comment]
I so desperately pine for the day that neither side gets off screen bombardment mechanics.
When one hand bears a thorn I've never once thought to drive a stake into the other.

**Comment by Arturlang** [comment]
Technically, the dragon will leave a shadow, so you can, in fact, see it flying around

**Comment by Coastfront** [comment]
> Technically, the dragon will leave a shadow, so you can, in fact, see it flying around @Coastfront

I still maintain that every single form of air support/bombardment mechanic is horrendous and boring in 9/10 applications of it.

**Comment by chizzy376** [comment]
> > Technically, the dragon will leave a shadow, so you can, in fact, see it flying around @Coastfront
> 
> I still maintain that every single form of air support/bombardment mechanic is horrendous and boring in 9/10 applications of it.

then maybe this pr isn't for you

**Comment by Coastfront** [comment]
> > > Technically, the dragon will leave a shadow, so you can, in fact, see it flying around @Coastfront
> > 
> > 
> > I still maintain that every single form of air support/bombardment mechanic is horrendous and boring in 9/10 applications of it.
> 
> then maybe this pr isn't for you

I genuinely cannot believe we have reached a point that on an open-scource codebase where public feedback exists that I have just  been told that a pr that would massively affect the balancing of the game is "not for me"

**Comment by Arturlang** [comment]
Both of you, please, unless you have specific ways of improving or clear reasons why this shouldn't be added, please discuss it elsewhere, a reason better than it's just boring.

**Comment by Losenis** [comment]
Out of all the reasons for any PR in TGMC to be told no, just being boring is a compliment in comparison

**Comment by Coastfront** [comment]
Alright then criticism beyond it is boring, which this criticism also applies to normal cas:
As for both of these, cas & xeno cas is just a thing that happens. (this also applies to mortar and howie aswell but atleast wraiths are good at dealing with those.)
It is area that explodes/lights-on-fire/whathaveyou that you get out of. Thats it. 
So far as what I can tell from the design doc, marines cant actually attack it while its in its funny cas sight mode, same as xenos cant actually do anything about the cas plane taking off and bombing them.
Same thing as-well. Cas cutting off xeno retreats, the "counter play" xenos get to a cas trying to cut them off is A. wipe the marines before you die from being cutoff 
B. cut your losses and run back to safety before the cas bomb/stun/giantfierylaser is at a solid risk of nailing you before you can retreat.
I say its boring as a simplification of this because that is the crystalization of every kind of cas. Its a thing that happens and the "counter play" to it is to not be where it happens. Normal marine CAS is the worst offender of this soley by virtue of the fact that xenos have to unironically just fucking win to make cas not be an issue anymore.

Boiler, mortar, howie, are atleast slightly less egregious offenders of this by virtue of "counters" to them existing (flanking, wraithing the mortar/howie, sniper for boiler, even though all of the counters are mediocre at best.) whereas dragon, so far as what I can tell from the design doc, could theoretically just have a drone setup backline completely walled off spaces to sit in to hide from flankers while they heal.
I'll admit, dragon is still not as cancerous as normal CAS but it doesen't change that IMO every type of bombardment mechanic in the game currently is poorly done, and in the case of things like CAS and CAS adjacent content, there isn't really a "good" way to make them fun, because they'll always feel detached from the actual gameplay loop of the game and just a force that "happens" to the opposing side.

**Comment by Goblinu** [comment]
> _The AIM-224 is the latest in air to air missile technology. Earning the nickname of 'Widowmaker'_
> 
> SOOOoooo.... The Condor can shoot them down right?

can xenos shoot down CAS?

**Comment by Arturlang** [comment]
@Coastfront Can you propose ways to counter the dragon? Currently using marine CAS isn't too practical, unless I just make fragmentation specifically work for hitting the dragon

**Comment by Coastfront** [comment]
> @Coastfront Can you propose ways to counter the dragon? Currently using marine CAS isn't too practical, unless I just make fragmentation specifically work for hitting the dragon

and theres the immediate issue with cas and cas adjacent mechanics, there isnt a "good" way to add a counter without having to add specific content soley for countering it.

If it was feasible, which it likely isn't, it'd probably just be preferable to add being able to "shoot up" at it's shadow, causing bullets to hit it on a delay so that it isnt just a "get fucked" when its in cas mode, but that is probably a nightmare to code.

**Comment by iforgotmeword** [comment]
just my shotgun idea on how T-4 should work because not going to lie king and dragon working together sounds kind of sounds OP(and fun) but the dragon is never going to be commonplace as it is stuck at 16-14 Xenos (that is about 30% of a 50-player game that is red eye blood cult numbers)

my (dumb) idea is to make it so the queen can buy T-4 towers for like 625 points allowing one Xeno to evo to a T-4 BUT at 12 Xenos or so. Xenos gets a T-4 tower for FREE and at like IDK 20 Xenos, the Xenos get another free T-4 tower just bypassing the need to use points for more T-4 towers. 

but T-4 towers have a period of growing time of like 2 mins when the T-4 tower is killed nothing happens UNLESS the T-4 dies then another T-4 slot doesn't open. however, if the Xenos has numbers needed for free towers the xenos can just make another one for free.

EX: 17 Xenos along with King and Dragon, droppod marine drops into the back line and kills both T-4 towers both King and Dragon dies at the front line there are now 15 Xenos the queen is able to make a free T-4 tower for free and after the growing period a Xeno is able to grow into a Dragon however due to lack of points queen can't make another tower so no King

this would kind of be a buff (closer to a rework) as Xenos can now if wanted get a T-4 without getting to the number count needed for said T-4 (king)

**Comment by Tailiik** [comment]
> Technically, the dragon will leave a shadow, so you can, in fact, see it flying around @Coastfront

No sunlight during HvX ever...but shadow? :thonk:

**Comment by novaepee** [comment]
I personally can't wait to see how dragon deal with FOB siege and backline, especially corpsmen healing marines.

**Comment by Araulius** [comment]
> @Coastfront Can you propose ways to counter the dragon? Currently using marine CAS isn't too practical, unless I just make fragmentation specifically work for hitting the dragon

you mean like how there's no counter play against the condor?

**Comment by Aporhtonoma** [comment]
touhou fights between dragon and cas when

**Comment by iforgotmeword** [comment]
> > @Coastfront Can you propose ways to counter the dragon? Currently using marine CAS isn't too practical, unless I just make fragmentation specifically work for hitting the dragon
> 
> you mean like how there's no counter play against the condor?

here is how you counterplay the condor and fire support use the fact maps are 50% caves and camp in said caves till they disconnect from boredom

**Comment by Oscari-pog** [comment]
dragon xeno
no wings

**Comment by Moggram** [comment]
> 
> TODO, not sure how I'd implement it, ideas?

> 
> can xenos shoot down CAS?


Here you go.


Working on it

**Comment by Arturlang** [comment]
> 

Um, if I were to make a whole new z level for aerial combat, I'd need sprites for the plane and a whole new design  doc to even figure out how to do that

**Comment by SilencedMP5A5** [comment]
[skyrim combat music begins]

**Comment by Al-1ce** [comment]
Aerial comnat is one of those things that _sounds cool _ but is actually fucking stupid as shit.

I think the initial design of this was fine as it is. Dragon could serve to replace what King was - costing a lot of psi-points to be able to spawn.

**Comment by Al-1ce** [comment]
> Aerial comnat is one of those things that _sounds cool_ but is actually fucking stupid as shit.
> 
> I think the initial design of this was fine as it is. Dragon could serve to replace what King was - costing a lot of psi-points to be able to spawn.

**Comment by Redrover1760** [comment]
> Aerial comnat is one of those things that _sounds cool_ but is actually fucking stupid as shit.
> 
> I think the initial design of this was fine as it is. Dragon could serve to replace what King was - costing a lot of psi-points to be able to spawn. It should _never_ be a guarenteed thing based on population like what King currently is.

The best concept my brain can come up with is a limited dedicated ability that disables the other CAS user for a limited period of time. The Dragon can disable the condor for a minute when its flying for a decent plasma cost and a nuts cooldown, like 11 minutes, or perhaps requiring a psy point cost to fire, even, and the condor can do likewise with a dedicated expensive (250 point) missile that also has to give warning to the Dragon for it to have time to land safely at home instead of in the middle of a bunch of marines. Also, both CAS users know if the other is active naturally. Both abilities only disable the CAS modes, essentially acting as a precision strike to allow xenos or marines to capitalize on a short timeframe where it is guaranteed that the enemy's CAS isnt active and to try and make significant progress with the minute or so of CAS freedom.

Of course, having a role that knows if CAS is active or not also helps with counterplaying CAS, both CAS users have to refuel here and so that means that the Dragon can spot that there's no Condor operational due to refueling and call that out and likewise for mini oppertunities, if CAS is really acting all that oppressive.

**Comment by Al-1ce** [comment]
> Or you could have anti air zones created by an emplacement of some kind that can't be undeployed by both sides that ban enemy CAS that can be destroyed or something.


Again, all of this may _sound_ very cool but I'd imagine would be very jank/unfun for either side.

**Comment by Arturlang** [comment]
> my main issue I can see is that the dragon would bomb the xeno maze, also maby adding a sort of smoke ability would be fun to give some help when there are large open fields. since those situations tend to be very stale mate inducing
> 
> also idea MABY add a anti air gun dragable so marines can atleast secure 1 forward camp/FOB maby.

Check the doc, there's already a smoke-ish ability for primo

**Comment by chizzy376** [comment]
Condor vs Dragon dogfight?

**Comment by Arturlang** [comment]
> Condor vs Dragon dogfight?

Probably for another PR, sorry!

**Comment by zzurge** [comment]
> > @Coastfront Can you propose ways to counter the dragon? Currently using marine CAS isn't too practical, unless I just make fragmentation specifically work for hitting the dragon
> 
> you mean like how there's no counter play against the condor?

in defense of the condor,it requires flare and marine coordination to even be remotely effective,and it also generally poses a significant risk to marines in and of itself. meaning that in the process of doing fire missions,it actually tends to kill more marines than xenos. moreover its gotten extremely difficult to do cas on the xeno backline. im interested to see if a "cas you can manually control" is going to be a bit too powerful

File: code/modules/mob/living/carbon/xenomorph/castes/dragon/castedatum_dragon.dm [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+
+	// *** Speed *** //
+	speed = 0.3
+
+	// *** Plasma *** //
+	plasma_max = 900
+	plasma_gain = 40
+
+	// *** Health *** //
+	max_health = 450
```
Comment (Pariah919) [inline code/modules/mob/living/carbon/xenomorph/castes/dragon/castedatum_dragon.dm:22]: 450 to 550 to 600 to 700??? Why is this scaling so hard?

Set it to 550/600/650/700.

Comment (Arturlang) [reply code/modules/mob/living/carbon/xenomorph/castes/dragon/castedatum_dragon.dm:22]: Probably wise! I was mostly going on starfire's suggestions, I'll change it like you suggested

**Comment by Mannybrado** [comment]
Looks cool.
Hope marines won't be able to see the area of effect radius thing, though, like how xenos can't see it for CAS.

**Comment by Arturlang** [comment]
New sprites for the king image

**Comment by Joe13413** [comment]
alright, i just saw some xenonauts aerial combat and i thought that it might be good to post here as a potential idea of condor vs dragon dogfighting, should that ever be coded. heres a professionally drawn picture of sorts on what the combat MAY look like
retard idea oh yeh

Heres a youtube link on xenonauts aerial combat 
https://www.youtube.com/watch?v=kbX77G9x2_c&ab_channel=Dranak
[DISCLAIMER] this video just gave me a few ideas that might translate well into TGMC, never played xenonauts and never coded before. Not sure how viable this all is but it just seems like a good starting point on what aerial combat may be

**Pull request closed without merging**